server/grpc_service.go

```diff
@@ -310,26 +340,55 @@ errHandling:
 	}
 }
 
-func (s *GrpcServer) processTSORequests(forwardStream pdpb.PD_TsoClient, requests []*tsoRequest) error {
+type tsoResp interface {
+	GetTimestamp() *pdpb.Timestamp
+}
+
+func (s *GrpcServer) processTSORequests(forwardStream pdpb.PD_TsoClient, forwardMCSStream tsopb.TSO_TsoClient, requests []*tsoRequest) error {
 	start := time.Now()
 	// Merge the requests
 	count := uint32(0)
 	for _, request := range requests {
 		count += request.request.GetCount()
 	}
-	req := &pdpb.TsoRequest{
-		Header: requests[0].request.GetHeader(),
-		Count:  count,
-		// TODO: support Local TSO proxy forwarding.
-		DcLocation: requests[0].request.GetDcLocation(),
-	}
-	// Send to the leader stream.
-	if err := forwardStream.Send(req); err != nil {
-		return err
+	var (
+		resp tsoResp
+		err  error
+	)
+	if forwardStream != nil {
```

Comment:
Will both forwardStream and forwardMCSStream not be nil?

Reply:
Line 281 ensure one of them will be not nil